pkg/providers/snow/apibuilder.go

```diff
@@ -135,11 +137,46 @@ func SnowCluster(clusterSpec *cluster.Spec) *snowv1.AWSSnowCluster {
 				Host: clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host,
 				Port: 6443,
 			},
+			IdentityRef: &snowv1.AWSSnowIdentityReference{
+				Name: CredentialsSecretName(clusterSpec),
+				Kind: snowv1.SecretKind,
```

Comment:
I don't love this. If in the future we want to support more types, we need to change two places. If we want to hardcode this, then I would vote to remove the `kind` field from our spec and just take a name.

I would expect this to pass-through whatever is set in the eks-a spec, since we already validate that it can only be secrets.